app/actors/hyrax/actors/base_actor.rb

```diff
@@ -113,6 +114,15 @@ def remove_blank_attributes!(attributes)
         def multivalued_form_attributes(attributes)
           attributes.select { |_, v| v.respond_to?(:select) && !v.respond_to?(:read) }
         end
+
+        def valkyrie_save(resource:)
+          permissions = resource.permission_manager.acl.permissions
+          resource    = Hyrax.persister.save(resource: resource)
+
+          resource.permission_manager.acl.permissions = permissions
```

Comment:
I notice that you get permissions from the resource, save the resource, reapply the permissions to the resource and then save the permissions.

Is it because the permissions do not last through a resource being saved?

Reply:
> Is it because the permissions do not last through a resource being saved?

not exactly. the object that `resource` references on L119 still has the permissions (until it's garbage collected). but we've set `resource = Hyrax.persister.save(resource: resource)`, and *this* new object knows nothing about unsaved permissions.

in general, it's desirable to just build the permissions and save them on a completely different lifecycle from `resource`, but since this is the first time `resource` gets saved, we need to save the resource first.

there might be a more illustrative implementation for this... i could think on it